src/player/player_update.cpp

```diff
@@ -596,18 +598,48 @@ void UpdatePlayer()
                     Player[A].CanFly2 = false;
                     Player[A].RunCount = 0;
                     Player[A].SpinJump = false;
-                    if(Player[A].Controls.Left)
-                        Player[A].Location.SpeedX = -1.5;
-                    else if(Player[A].Controls.Right)
-                        Player[A].Location.SpeedX = 1.5;
-                    else
-                        Player[A].Location.SpeedX = 0;
-                    if(Player[A].Controls.Up && Player[A].Vine > 2)
-                        Player[A].Location.SpeedY = -2;
-                    else if(Player[A].Controls.Down)
-                        Player[A].Location.SpeedY = 3;
+                    if(!g_compatibility.player_faster_climbing)
+                    {
+                        if(Player[A].Controls.Left)
+                            Player[A].Location.SpeedX = -1.5;
+                        else if(Player[A].Controls.Right)
+                            Player[A].Location.SpeedX = 1.5;
+                        else
+                            Player[A].Location.SpeedX = 0;
+                        if(Player[A].Controls.Up && Player[A].Vine > 2)
+                            Player[A].Location.SpeedY = -2;
+                        else if(Player[A].Controls.Down)
+                            Player[A].Location.SpeedY = 3;
+                        else
+                            Player[A].Location.SpeedY = 0;
+                    }
                     else
-                        Player[A].Location.SpeedY = 0;
+                    {
+                        if(Player[A].Controls.Left)
+                            if(!Player[A].Controls.Run)
+                                Player[A].Location.SpeedX = -1.5;
+                            else
+                                Player[A].Location.SpeedX = -3;
+                        else if(Player[A].Controls.Right)
```

Comment:
You better to add braces around these blocks. Without braces, this looks like Python...

Reply:
Example:
```cpp
if(xxx)
{
    if(yyy)
        do();
    else
        do2();
}
else if(zzz)
{
    if(qqq)
        do3();
    else
        do4();
}
```